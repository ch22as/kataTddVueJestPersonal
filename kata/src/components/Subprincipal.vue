<template>
  <div class="container">
      <h1>Subprincipal</h1>
      <img src="https://picsum.photos/200/300" alt="">
  </div>
</template>

<script>
export default {
    name: 'Subprincipal'

}
</script>

<style>

</style>