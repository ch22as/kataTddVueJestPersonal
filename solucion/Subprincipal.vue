<template>
  <div class="container subprincipal">
    <h1>Subprincipal</h1>
    <img src="https://picsum.photos/200/300" alt="Imagen random" />
    <Worker/>
  </div>
</template>

<script>
import Worker from './Worker.vue';

export default {
    name: 'Subprincipal',
    components: {
        Worker
    }

}
</script>

<style>
.subprincipal {
  background: blue;
  padding: 1%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
</style>