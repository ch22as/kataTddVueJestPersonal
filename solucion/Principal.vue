<template>
  <div class="container">
      <Subprincipal/>
  </div>
  

</template>

<script>
import Subprincipal from '@/components/Subprincipal.vue'

export default {
    name:'Principal',
    components: {
      Subprincipal
    }
}
</script>

<style>

</style>