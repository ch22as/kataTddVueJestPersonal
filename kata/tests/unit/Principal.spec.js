import { mount } from '@vue/test-utils';
import Principal from '@/views/Principal';
import Subprincipal from '@/components/Subprincipal.vue'

const wrapper = mount(Principal);

describe('Principal', () =>{
    it('it renders the component', () =>{
       

        expect(wrapper.html()).toMatchSnapshot();
    });

    it(' it have the components?', () => {
        const subprincipal = wrapper.findAllComponents(Subprincipal);

        expect(subprincipal.exists()).toBe(true);
    })
});


