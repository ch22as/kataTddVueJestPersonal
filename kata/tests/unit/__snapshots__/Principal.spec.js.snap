// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Principal it renders the component 1`] = `
<div class="container">
  <div class="container subprincipal">
    <h1>Subprincipal</h1> <img src="https://picsum.photos/200/300" alt="Imagen random">
    <div class="container worker"><input type="text" placeholder="Introduzca texto" name=""> <button>Enviar</button></div>
  </div>
</div>
`;
